_posts/2020-03-09-o-openid-connect-slow-kilka.md

```diff
@@ -0,0 +1,106 @@
+[**"OAuth 2.0 – jak działa / jak testować / problemy bezpieczeństwa"** (link](https://sekurak.pl/oauth-2-0-jak-dziala-jak-testowac-problemy-bezpieczenstwa/))  
+autorstwa Marcina Pioska z portalu [Sekurak](https://sekurak.pl/), w którym została opisana terminologia, sposoby pozyskiwania tokenu oraz zasada działania standardu.
+
+O JWT natomiast możemy przeczytać w dokumencie [RFC7519](https://tools.ietf.org/html/rfc7519).
+
+Dlaczego więc do integracji naszej aplikacji z serwerem uwierzytelniania Keycloak użyliśmy OpenID Connect?
+
+## Uwierzytelnianie a autoryzacja
+
+Jeśli mówimy o zabezpieczeniu zasobów i dostępem do nich, mówimy o takich pojęciach jak **uwierzytelnianie** (ang. authentication) oraz **autoryzacja** (ang. authorization).
+
+- `uwierzytelnianie` to proces polegający na potwierdzeniu tożsamości, czyli w skrócie - *kim jestem?*;
+- `autoryzacja` to proces nadawania uprawnień (dostępu do zasobu), czyli w skrócie - *co mogę zrobić?*.
+
+OAuth 2.0 według oficjalnej dokumentacji nie powinien służyć do uwierzytelniania, a jedynie do autoryzacji ([źródło](https://oauth.net/articles/authentication/)):
+> OAuth 2.0 is not an authentication protocol.
+
+Jeśli potrzebujemy mechanizmu pozwalającego na poprawne zaimplementowanie uwierzytelniania, z pomocą przychodzi OpenID Connect.
+
+## OpenID Connect
+
+OpenID Connect jest prostą warstwą tożsamości opartą na OAuth 2.0.  
+Umożliwia klientom weryfikację tożsamości użytkownika końcowego na podstawie uwierzytelnienia przeprowadzonego przez serwer autoryzacji, a także uzyskanie podstawowych informacji o profilu użytkownika. Rozszerza OAuth 2.0, umożliwiając **uwierzytelnianie stowarzyszone** (ang. federated authentication):
+
+- **Federated Authentication** - użytkownik loguje się do serwisu Spotify przy użyciu konta na portalu Facebook (`OpenID Connect`);
+- **Delegated Authorization** - Spotify próbuje uzyskać dostęp do listy znajomych na Facebooku, aby zaimportować ją do swojej bazy danych (`OAuth 2.0`).
+
+Przepływ procesu jest podobny do OAuth 2.0, ale dodatkowo w procesie bierze udział **ID Token**.
+
+## ID Token
+
+ID Token przypomina koncepcję dowodu osobistego w formacie JWT, podpisanym przez dostawcę OpenID (OP). Spełnia założenia standardu JWT, więc składa się z nagłówka, zawartości oraz
+sygnatury. Jego zawartość może więc wyglądać następująco:
+
+```json
+{
+  "jti": "e78b5f41-e769-4353-8874-44302f4a17c3",
+  "exp": 1583205992,
+  "nbf": 0,
+  "iat": 1583169992,
+  "iss": "http://localhost:8180/auth/realms/SpringBootAngular",
+  "aud": "developer",
+  "sub": "c3c3755a-e499-4782-b119-a19bede0ace8",
+  "typ": "Serialized-ID",
+  "nonce": "02c0b033-bfac-4030-a317-c18aec3cb2db",
+  "auth_time": 1583169991,
+  "acr" : "1",
+  "session_state": "8b66127b-4474-41bd-8e36-5d18286df73f",
+  "state_checker": "HZZmC4no-TEqiCf31Mk1MtONDqyxkk81ZXZCwANQb9Y",
+  "name": "Jan Nowak",
+  "email": "jan.nowak@example.pl"
+}
+```
+
+**Jego główne funkcje to m.in.:**
```

Comment:
Jeśli lista zaczyna się od wstępu "jego główne funkcje, to", to w kolejnych punktach powinny być równoważniki zdań, np. "potwierdzanie tożsamości [...]", "określanie organu [...]". Trzeba zmienić albo wstęp do listy, albo samą listę